Doc/library/os.rst

```diff
@@ -3780,6 +3780,174 @@ features:
+
+   .. availability:: Linux >= 2.6.27 with glibc >= 2.8
+
+   .. versionadded:: 3.13
+
+
+
+.. function:: timerfd_settime_ns(fd, flags, interval_ns, value_ns)
+
+   Start/stop timer for an timerfd file descriptor.
+   See man page :manpage:`timerfd_settime(2)` for more information.
+
+   ``interval_ns`` coresponds to ``it_interval`` in ``struct itimerspec`` and ``value_ns`` coresponds to
+   ``it_value`` in ``struct itimerspec``. They are both in nano second unit and types are long long.
+
+   ``interval`` is calculated like ``it_interval.tv_sec * 10**9 + it_interval.tv_nsec``,
+   ``value`` is calculated like ``it_interval.tv_sec * 10**9 + it_interval.tv_nsec`` internally.
+
+   .. availability:: Linux >= 2.6.27 with glibc >= 2.8
+
+   .. versionadded:: 3.13
+
+
+.. function:: timerfd_gettime(fd)
+
+   Returns a tuple ``interval`` which and ``value`` which is the time until next expiration.
+   See man page :manpage:`timerfd_gettime(2)` for more information.
+
+   ``interval`` coresponds to ``it_interval`` in ``struct itimerspec`` and ``value`` coresponds to
+   ``it_value`` in ``struct itimerspec``. They are both in seconds unit and types are double.
+
+   ``interval`` is calculated like ``it_interval.tv_sec + it_interval.tv_nsec * 1e-9``,
+   ``value`` is calculated like ``it_interval.tv_sec + it_interval.tv_nsec * 1e-9`` internally.
+
+   .. availability:: Linux >= 2.6.27 with glibc >= 2.8
+
+   .. versionadded:: 3.13
+
+
+.. function:: timerfd_gettime_ns(fd)
+
+   Returns a tuple ``interval`` which and ``value`` which is the time until next expiration.
+   See man page :manpage:`timerfd_gettime(2)` for more information.
+
+   ``interval`` coresponds to ``it_interval`` in ``struct itimerspec`` and ``value`` coresponds to
+   ``it_value`` in ``struct itimerspec``. They are both in nano second unit and types are long long.
+
+   ``interval`` is calculated like ``it_interval.tv_sec * 10**9 + it_interval.tv_nsec``,
+   ``value`` is calculated like ``it_interval.tv_sec * 10**9 + it_interval.tv_nsec`` internally.
+
+   .. availability:: Linux >= 2.6.27 with glibc >= 2.8
+
+   .. versionadded:: 3.13
+
+.. data:: TFD_CLOEXEC
+
+   Set close-on-exec flag for new :func:`timerfd_create` file descriptor.
+
+   .. availability:: Linux >= 2.6.27 with glibc >= 2.8
+
+   .. versionadded:: 3.13
+
+.. data:: TFD_NONBLOCK
+
+   Set :const:`O_NONBLOCK` status flag for new :func:`timerfd_create` file
+   descriptor.
+
+   .. availability:: Linux >= 2.6.27 with glibc >= 2.8
+
+   .. versionadded:: 3.13
+
+.. data:: TFD_TIMER_ABSTIME
+
+   Set :const:`TFD_TIMER_ABSTIME` flags for :func:`timerfd_settime` or
+   :func:`timerfd_settime_ns`.
+
+   .. availability:: Linux >= 2.6.27 with glibc >= 2.8
+
+   .. versionadded:: 3.13
+
+
+.. data:: TFD_TIMER_CANCEL_ON_SET
+
+   Set :const:`TFD_TIMER_CANCEL_ON_SET` flags for :func:`timerfd_settime` or
+   :func:`timerfd_settime_ns`.
+
+   .. availability:: Linux >= 2.6.27 with glibc >= 2.8
+
+   .. versionadded:: 3.13
```

Comment:
You can collect these constants under one entry. See for example the `SEEK_` or `PRIO_` constants.

Reply:
@erlend-aasland 
Fixed.